### ISSUE TITLE
Allow files without top module

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#953]
For throw-away Agda files, I would like to be able to omit the 
```agda
  module JustAThrowAwayFileButModuleNameMustMatchFileNameExactly where
```
top-level module declaration.

Use cases:
  - little bits of Agda code in a presentation
  - test suite
  - playing, experimentation and demonstration

If no module name is encounterd before the first non-import statement, just the (unqualified) file name is taken as module name.

Original issue reported on code.google.com by `andreas....@gmail.com` on 12 Nov 2013 at 1:59

**Comment by andreasabel** [comment]
Fix for this issue introduced #1388 in commit https://github.com/agda/agda/commit/f30432239881536f92f50bc048307453ca528279.
#1388 fixed in https://github.com/agda/agda/commit/470dbcf7666a1fbd379406f60f678e36e9b5996e.

**Comment by andreasabel** [comment]
I admit that dropping the module header altogether did not pass the test of time.  It makes recognizing the top-level module too context-dependent.  Omitting the name and writing `module _ where` however seems stable.
